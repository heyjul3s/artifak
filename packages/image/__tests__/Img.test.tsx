import React from 'react';
import { render } from '@testing-library/react';
import { Img } from '../';

describe('Img', () => {
  it('should render an img element with src and alt defined', () => {
    const { asFragment } = render(
      <Img src="https://source.unsplash.com/random/10x10" alt="hello world" />,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render an img element with fallbackSrc value as src', () => {
    const { asFragment } = render(
      <Img
        src="https://sourcelaopj.com/random/10x10"
        fallbackSrc="https://source.unsplash.com/random/10x10"
      />,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
