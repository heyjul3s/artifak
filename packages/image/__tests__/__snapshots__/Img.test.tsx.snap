// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img should render an img element with fallbackSrc value as src 1`] = `
<DocumentFragment>
  <img
    class="artifak__sc-6om8fa-0 iwKMLk"
    src="https://sourcelaopj.com/random/10x10"
  />
</DocumentFragment>
`;

exports[`Img should render an img element with src and alt defined 1`] = `
<DocumentFragment>
  <img
    alt="hello world"
    aria-label="hello world"
    class="artifak__sc-6om8fa-0 iwKMLk"
    src="https://source.unsplash.com/random/10x10"
  />
</DocumentFragment>
`;
